### ISSUE TITLE
Expression: Cannot use boolean value directly in conditional

### ISSUE THREAD

**Issue by Roy-043** [FreeCAD/FreeCAD#15558]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

You cannot use a Boolean value directly as the condition in a conditional expression.

You have to use this syntax:
`VarSet.MyBool == 1 ? 10 mm : 15 mm`

Where I would expect this to be valid as well:
`VarSet.MyBool ? 10 mm : 15 mm`

And this should also work IMO:
`1 ? 10 mm : 15 mm`

Documentation:
https://wiki.freecad.org/Expressions#Conditional_expressions

### Full version info

```shell
OS: Windows 8 build 9600
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38231 (Git)
Build type: Release
Branch: main
Hash: 29889bfd9a7ba0f0496f2e2699d454ea5fb5a05a
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: C/Default (C) [ OS: Dutch/Netherlands (nl_NL) ]
Installed mods:
```

### Subproject(s) affected?

Core

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by maxwxyz** [comment]
@Roy-043 is this the same as https://github.com/FreeCAD/FreeCAD/issues/8008 ?

**Comment by Roy-043** [comment]
I did find #8008 before creating this issue. It is obviously related but different.

**Comment by mnesarco** [comment]
@maxwxyz 
#22506 will not close this issue, it solves a different problem.

**Comment by mnesarco** [comment]
Hello @maxwxyz #22630 is intended to close this issue.

**Comment by jffmichi** [comment]
#22630 is introducing an `if` function as a workaround for this issue. This issue should not be closed by that PR.

**Comment by mnesarco** [comment]
> [#22630](https://github.com/FreeCAD/FreeCAD/pull/22630) is introducing an `if` function as a workaround for this issue. This issue should not be closed by that PR.

Yes but, I think that `if` function is a better solution for conditionals, it is more familiar to normal users as the FreeCAD expressions prefer excel syntax over c syntax.

**Comment by jffmichi** [comment]
So if the trenary operator is removed, this issue can be closed. As long as the trenary operator exists, this issue remains valid. (But I think noone is seriously proposing that here).

**Comment by mnesarco** [comment]
The ternary operator should not be removed, as we want to keep backwards compatibility and do not break existing models. It can be deprecated (supported with a warning).

Changing the ternary operator to support direct use of `cond` rule as `exp` requires some refactoring of the base grammar. Its not impossible of course but needs extensive and well designed tests because it can be tricky to ensure the semantic remains unchanged.

**Comment by mnesarco** [comment]
I agree that #22630 provides an alternative conditional mechanism, not a direct solution to the ternary operator limitation itself (this issue)

**Comment by jffmichi** [comment]
I would argue against deprecating it but that's just a personal preference (C++ > Excel !!!). And if it's there it would be nice if it's fully supported as someone using it would reasonably expect. That's why I'm arguing to keep this open until someone finds the time to properly fix the trenary. Your introduction of the `if` function still is a great addition. I don't think there's harm in supporting both syntaxes.
